server/src/main/java/org/opensearch/index/query/QueryShardContext.java

```diff
@@ -538,6 +548,64 @@ private ParsedQuery toQuery(QueryBuilder queryBuilder, CheckedFunction<QueryBuil
         }
     }
 
+    public ParsedQuery toStarTreeQuery(
+        CompositeIndexFieldInfo starTree,
+        CompositeDataCubeFieldType compositeIndexFieldInfo,
+        QueryBuilder queryBuilder,
+        Query query
+    ) {
```

Comment:
Does this method (and `getStarTreePredicates`) need to live in `QueryShardContext`? I think they could both be static methods in a utility class.